api/src/main/java/org/apache/iceberg/expressions/ExpressionVisitors.java

```diff
@@ -55,6 +55,14 @@ public <T> R predicate(BoundPredicate<T> pred) {
     public <T> R predicate(UnboundPredicate<T> pred) {
       return null;
     }
+
+    public <T, C> R aggregate(BoundAggregate<T, C> agg) {
+      throw new UnsupportedOperationException("aggregate is not supported");
+    }
+
+    public <T> R aggregate(UnboundAggregate<T> agg) {
+      return null;
```

Comment:
I think this should also throw `UnsupportedOperationException`

Reply:
Changed. Thanks!